opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -741,11 +736,26 @@ DenseMatrix transposeDenseMatrix(const DenseMatrix& M)
             if (recreate_solver || !flexibleSolver_) {
                 if (isParallel()) {
 #if HAVE_MPI
-                    assert(noGhostMat_);
-                    flexibleSolver_.reset(new FlexibleSolverType(getMatrix(), *comm_, prm_, weightsCalculator));
+                    if (useWellConn_) {
+                        assert(noGhostMat_);
+                        using ParOperatorType = Dune::OverlappingSchwarzOperator<Matrix, Vector, Vector, Comm>;
+                        linearOperatorForFlexibleSolver_ = std::make_unique<ParOperatorType>(getMatrix(), *comm_);
+                        flexibleSolver_ = std::make_unique<FlexibleSolverType>(*linearOperatorForFlexibleSolver_, *comm_, prm_, weightsCalculator);
+                    } else {
+                        if (!ownersFirst_) {
+                            OPM_THROW(std::runtime_error, "In parallel, the flexible solver requires "
+                                      "--owner-cells-first=true when --matrix-add-well-contributions=false is used.");
```

Comment:
Learned something new here.

Reply:
This could be changed, but I chose to only allow the ...GhostLast version of the operator, since it is superior in performance and simpler in implementation. If we want to allow `--owner-cells-first=false` we will have to add block essentially identical to the below where we make a `WellModelMatrixAdapter`. Personally, I do not plan on ever running without owner cells first though... (You may convince me otherwise if that is a bad idea.)